lib/galaxy/tool_util/deps/container_resolvers/mulled.py

```diff
@@ -269,21 +269,17 @@ def cached_name(cache_key):
         tags = mulled_tags_for(namespace, target.package_name, resolution_cache=resolution_cache)
 
         if tags:
-            if target_version:
-                for tag in tags:
-                    if '--' in tag:
-                        version, build = split_tag(tag)
-                    else:
-                        version = tag
-                        build = None
-                    if version == target_version:
-                        name = "%s:%s" % (target.package_name, version)
-                        if build:
-                            name = "%s--%s" % (name, build)
-                        break
-            else:
-                version, build = split_tag(tags[0])
-                name = "%s:%s--%s" % (target.package_name, version, build)
+            for tag in tags:
+                if '--' in tag:
+                    version, build = split_tag(tag)
+                else:
+                    version = tag
+                    build = None
+                if target_version and version == target_version:
+                    name = "%s:%s" % (target.package_name, version)
+                    if build:
+                        name = "%s--%s" % (name, build)
```

Comment:
```suggestion
                if '--' in tag:
                    version, _ = split_tag(tag)
                else:
                    version = tag
                if target_version and version == target_version:
                    name = "%s:%s" % (target.package_name, tag)
```
But I think it would make sense to have `mulled_tags_for` return `Target` named tuples so we don't have all these sprawling ad-hoc parsing of image tags.